src/error.rs

```diff
@@ -65,6 +65,20 @@ impl Error {
     /// custom errors.
     pub const CUSTOM_START: u32 = (1 << 31) + (1 << 30);
 
+    #[allow(dead_code)]
+    #[cold]
+    pub(super) fn new_os(code: u32) -> Self {
```

Comment:
> Could you add some docs for this method?

OK. But...

> pub(crate) fn from_os_code(code: impl TryInto<u32>) -> Self {

Unfortunately that would allow passing a `usize` or similar, which really shouldn't be implicitly converted. So I think we should not do that.

> I don't think we need to distinguish between ERRNO_NOT_POSITIVE and UNEXPECTED

I am OK with making that change.

Reply:
>Unfortunately that would allow passing a usize or similar, which really shouldn't be implicitly converted.

Since it's a private API, I think it's fine to allow this. If you want to make this more explicit, you could use `Error::from_os_code::<isize>(code)` at call sites. Either way, I do not insist on it.

Reply:
Well, I think the main goal here is to ensure that the result is well-defined, rather than to increase convenience.

I do hope this API can be improved further and then made public to replace the `From<NonZeroU32>` implementation as discussed elsewhere.

Reply:
I added the documentation, moved the `ERRNO_NOT_POSITIVE` logic back to where it belongs, and renamed the function to be more like the analogous `std::io::Error` function.